src/elements/review_by_id.jsx

```diff
@@ -1,11 +1,23 @@
-import { getReviewsFromId, getCommentsFromId } from "../utility/axios-request";
+import {
+  getReviewsFromId,
+  getCommentsFromId,
+  patchPlusVote,
+} from "../utility/axios-request";
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import ReviewCommentCard from "./reviews_comment_card";
 
 function ReviewById() {
   const reviewId = useParams();
   const number = Number(reviewId.id);
+  const [pressed, setPressed] = useState(false);
+  const [votes, setVotes] = useState(0);
+  useEffect(() => {
+    patchPlusVote(number, votes).then((response) => {
```

Comment:
at the moment this isn't optimistically rendering the votes, the button is disabled only while the patch request to the api goes through and the new vote count is only displayed after a successful patch. Can you optimistically render this so the new vote count is displayed straight away, but make it so that the user can't vote again while we wait for the request.